### PR TITLE
Add EncodeBase32() to Redbean

### PR DESCRIPTION
**Pull request by pkulchenko**
@jart, this implements EncodeBase32, but I tweaked the function from google-authenticator-libpam to (1) allow the alphabet to be configurable and (2) work with any power of 2. I added a couple of tests, so it seems to do what's needed. I don't need DecodeBase32, so it's not included, but let me know if you think it's worth adding for consistency.

### CODE REVIEW

File: tool/net/lfuncs.c [hunk 1]
```diff
@@ -602,6 +602,64 @@ int LuaEncodeLatin1(lua_State *L) {
...
 }
 
+int tobits(int b) {
+  int bits = 0; while (b && (b >>= 1)) bits++;
+  return bits;
+}
+
+// the next function is based on
+// https://github.com/google/google-authenticator-libpam/blob/master/src/base32.c
+// Copyright 2010 Google Inc.; Author: Markus Gutschke
```
Comment (jart) [inline tool/net/lfuncs.c:612]: Could you put it in its own file, with its own license header?

**Review by jart — CHANGES_REQUESTED** [review]
Sure having the decoder would be cool.

File: tool/net/lfuncs.c (hunk 1 above)
Comment (pkulchenko) [reply tool/net/lfuncs.c]: @jart, I reorganized all the code and added decoding (along with the documentation).

**Comment by pkulchenko** [comment]
@jart, I followed the same style/convention as is used in Base64 encoding/decoding with one exception: the decoding with the default alphabet accepts/skips whitespaces and dash, but stops at the first character outside of the alphabet (returning the produced result). I can change it to continue (by updating the default map), but I like it a bit better.

**Review by jart — APPROVED** [review]
Thank you!

**Pull request merged**